worlds/kdl3/Client.py

```diff
@@ -310,9 +310,12 @@ async def game_watcher(self, ctx) -> None:
                 return  # null, title screen, opening, save select, true and false endings
             game_state = await snes_read(ctx, KDL3_GAME_STATE, 1)
             current_hp = await snes_read(ctx, KDL3_KIRBY_HP, 1)
+            current_world = struct.unpack("H", await snes_read(ctx, KDL3_CURRENT_WORLD, 2))[0]
+            current_level = struct.unpack("H", await snes_read(ctx, KDL3_CURRENT_LEVEL, 2))[0]
             if "DeathLink" in ctx.tags and game_state[0] == 0x00 and ctx.last_death_link + 1 < time.time():
                 currently_dead = current_hp[0] == 0x00
-                await ctx.handle_deathlink_state(currently_dead)
+                message = deathlink_messages[self.levels[current_world][current_level - 1]]
```

Comment:
Since reads are not atomic, this may end up being garbage data and producing an index error. I'm assuming you are fine with it going to the except in line 416, but wanted to make sure.

Reply:
If a read fails, then it's fine to go to the except. If that's the case, it's not impossible the health read was also garbage and a deathlink shouldn't be sent.